### PR TITLE
Coreth 0.11.6 sync

### PR DESCRIPTION
**Pull request by ceyonur**
## Why this should be merged

Syncs coreth 0.11.6 changes

## How this works

```
$ git checkout -b coreth-0.11.6-sync
$ git remote add coreth git@github.com:ava-labs/coreth.git
$ git fetch --all
$ git diff v0.11.3 v0.11.6 > diff.patch
$ git apply diff.patch --reject --whitespace=fix
 ```

v0.11.3 was the last synced tag to subnet-evm (prior to this PR) 

After applying the patch, then iterate manually over rejected files. at this point I preferred iterating over coreth diffs in a separate window and checked differences manually. I.e open 2 windows one:
1- within coreth that shows diffs between v0.11.3 to v0.11.6 
2- subnet-evm 
and compare missing differences in folders with those rejected files. 

## How this was tested

UTs should cover this.

### CODE REVIEW

File: core/blockchain_test.go [hunk 1]
```diff
@@ -635,7 +635,7 @@ func TestUngracefulAsyncShutdown(t *testing.T) {
 // TestCanonicalHashMarker tests all the canonical hash markers are updated/deleted
 // correctly in case reorg is called.
 func TestCanonicalHashMarker(t *testing.T) {
-	var cases = []struct {
+	cases := []struct {
```
Comment (darioush) [inline core/blockchain_test.go:638]: nit: can we revert this change?

File: eth/filters/bench_test.go [hunk 2]
```diff
@@ -162,7 +162,7 @@ func benchmarkBloomBits(b *testing.B, sectionSize uint64) {
 
 //nolint:unused
 func clearBloomBits(db ethdb.Database) {
-	var bloomBitsPrefix = []byte("bloomBits-")
+	bloomBitsPrefix := []byte("bloomBits-")
```
Comment (darioush) [inline eth/filters/bench_test.go:165]: same nit here

File: core/blockchain_test.go (hunk 1 above)
Comment (ceyonur) [reply core/blockchain_test.go]: reverted

**Comment by darioush** [comment]
great work, couple nits:
https://github.com/ava-labs/subnet-evm/pull/441/files

**Review by darioush — APPROVED** [review]

File: plugin/evm/config.go [hunk 3]
```diff
@@ -15,37 +15,38 @@ import (
...
+	defaultWsCpuMaxStored                             = 0 // Default to no maximum WS CPU usage
+	defaultMaxBlocksPerRequest                        = 0 // Default to no maximum on the number of blocks per getLogs request
+	defaultContinuousProfilerFrequency                = 15 * time.Minute
+	defaultContinuousProfilerMaxFiles                 = 5
+	defaultRegossipFrequency                          = 1 * time.Minute
+	defaultRegossipMaxTxs                             = 16
+	defaultRegossipTxsPerAddress                      = 1
+	defaultPriorityRegossipFrequency                  = 1 * time.Second
+	defaultPriorityRegossipMaxTxs                     = 32
+	defaultPriorityRegossipTxsPerAddress              = 16
```
Comment (aaronbuchwald) [inline plugin/evm/config.go:41]: Can we migrate these to coreth as well?

**Review by aaronbuchwald — APPROVED** [review]
LGTM - thanks for putting this up 🙌

File: plugin/evm/config.go (hunk 3 above)
Comment (ceyonur) [reply plugin/evm/config.go:41]: you mean priority regossiping?
